### ISSUE TITLE
epochs?

### ISSUE THREAD

**Issue by tfor** [Mattral/Kolmogorov-Arnold-Networks#1]
In mnist.py there is this for loop (please pardon my possibly outdated verbiage as I started programming in Pascal and FORTRAN) given below between the lines of ====.

But epochs is not defined error is being thrown.  

To what value should epochs be set?

===================================================================================
for epoch in range(epochs):
    train_loss.reset_states()
    val_loss.reset_states()
    val_accuracy.reset_states()

    # Existing training and validation loop here ...

    # Append metrics after each epoch
    epoch_train_loss.append(train_loss.result().numpy())
    epoch_val_loss.append(val_loss.result().numpy())
    epoch_val_accuracy.append(val_accuracy.result().numpy())

    print(f'Epoch {epoch + 1}, Train Loss: {train_loss.result():.4f}, Validation Loss: {val_loss.result():.4f}, Validation Accuracy: {val_accuracy.result():.4f}')
===================================================================================

**Comment by Mattral** [comment]
Thank you and My apologies indeed there was an error as you have pointed out, I was not very active back then; now I am maintaining and now this repo has improved to production grade libraries (though we have some other issues to solve now)

you may try out the collab of how to use pip install kanx here
<a href="https://colab.research.google.com/github/Mattral/KANX/blob/main/notebooks/quickstart.ipynb"><img alt="Colab" src="https://img.shields.io/badge/Colab-train_in_2_to_5_min-F9AB00?style=for-the-badge&logo=googlecolab&logoColor=white"></a>

**Comment by Mattral** [comment]
we need to explicitly set a value for `epochs` before the loop. A common choice for MNIST is between **5 and 20 epochs**, depending on how long you want to train and how much accuracy you need.  

For example:

```python
epochs = 10  # train for 10 epochs
```

That will resolve the `NameError` and give you a reasonable training run. If you want faster experiments, set `epochs = 5`; for better accuracy, try `epochs = 20`.  

---

However please note that that old legacy implementation is archived into 'KAN_from_Scratch' directory and now please kindly proceed with the pip installable library.
